### DISCUSSION TITLE
2.x

### DISCUSSION THREAD

**Discussion by ryangjchandler** [ryangjchandler/orbit#109] [Ideas]
I'm currently working on version 2.x of Orbit. 1.x hasn't been out for long, but there are lots of improvements I want to make.

### Improve ergonomics

A lot of customisation right now takes place in individual methods on the model, i.e. changing the driver etc. My goal is to unify this all into a single method, similar to Spatie packages where you use an "options" object to configure your Orbital.

### Better performance

The performance of Orbit on larger datasets could be problematic. Right now, a single file change will cause the entire model cache to be updated and rewritten. This is unnecessary, a better way would be only updating the files that are newer than the cache itself.

### Better driver API

The driver API is weird now. Internally we use Laravel's manager class, but that's unnecessary. I'm going to be refining the API a lot to make it really simple to create custom drivers. This would make it really easy to create drivers for new file formats like CSV files, etc.

### Your suggestions

I'm looking for other ideas too. The list so far would make a 2.x release worthwhile in my opinion, but the more features the merrier!

**Comment by johncarter-** [comment]
~~Many to many relationships - could that be a thing?~~ Already possible 👌.

**Reply by ryangjchandler** [reply]
@johncarter I think it's technically possible now, but requires a custom pivot model... can't confirm for sure, since I haven't tried it myself.

**Reply by johncarter-** [reply]
Interesting. Ill have a look into it.
